### PR TITLE
Initialize prow support for ironic-standalone-operator

### PR DESCRIPTION
**Pull request by Rozzii**
This commit:
- Adds markdownlint and go mod tests for ironic-standalone-operator
- Adds basic prow branch protection for ironic-standalone-operator

### CODE REVIEW

File: prow/manifests/overlays/metal3/config.yaml [hunk 1]
```diff
@@ -196,6 +197,11 @@ branch-protection:
             main:
               required_status_checks:
                 contexts: ["test-centos-integration-main"]
+        ironic-standalone-operator:
+          branches:
+            main:
+              required_status_checks:
+                contexts: []
```
[the remark this answers is not in the document]
Comment (tuminoid) [reply prow/manifests/overlays/metal3/config.yaml:204]: You can add the whole section when there is something to add.

[the remark this answers is not in the document]
Comment (Rozzii) [reply prow/manifests/overlays/metal3/config.yaml:204]: I would like to keep it, I like that this is a more explicit description it shows that the repo is accounted for so next time when me or someone else wold like to add a "context" there is no need to wonder where to put it . If you really want to remove it I will do it but I prefer it this way.

[the remark this answers is not in the document]
Comment (tuminoid) [reply prow/manifests/overlays/metal3/config.yaml:204]: Not going to argue further, lets get this merged :)

**Review by tuminoid — COMMENTED** [review]
/approve

/hold
to give you sync control

**Review by smoshiur1237 — COMMENTED** [review]
/lgtm

**Pull request merged**